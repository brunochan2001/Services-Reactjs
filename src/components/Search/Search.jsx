import React, { useState } from 'react';
import { useDebounce } from 'react-use';
import TextField from '@mui/material/TextField';
import './index.css';

const Search = ({ services, setServicesData }) => {
  const [val, setVal] = useState('');
  const [,] = useDebounce(
    () => {
      SearchData(val);
    },
    400,
    [val]
  );

  const SearchData = value => {
    const lowerCaseValue = value.toLowerCase().trim();
    if (!lowerCaseValue) {
      setServicesData(services);
    } else {
      const searchServices = services.filter(item => {
        return Object.keys(item).some(key =>
          item[key].toString().toLowerCase().includes(lowerCaseValue)
        );
      });
      setServicesData(searchServices);
    }
  };

  return (
    <>
      <TextField
        fullWidth
        label="Buscador"
        id="fullWidth"
        className="search-input"
        onChange={({ currentTarget }) => {
          setVal(currentTarget.value);
        }}
      />
    </>
  );
};

export default Search;
