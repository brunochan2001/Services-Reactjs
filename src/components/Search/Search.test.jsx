import { render } from '@testing-library/react';
import Search from './Search';

describe('Pruebas en el componente <Search/>', () => {
  test('debe de hacer match con el snapshot', () => {
    const { container } = render(<Search></Search>);
    expect(container).toMatchSnapshot();
  });
});
