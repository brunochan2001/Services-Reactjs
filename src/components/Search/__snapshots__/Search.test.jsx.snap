// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <Search/> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root search-input css-wb57ya-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="false"
      for="fullWidth"
      id="fullWidth-label"
    >
      Buscador
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        id="fullWidth"
        type="text"
        value=""
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-1ftyaf0"
        >
          <span>
            Buscador
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
